### DISCUSSION TITLE
Will StateProvider be available soon in riverpod_generator?

### DISCUSSION THREAD

**Discussion by julienlebren** [rrousselGit/riverpod#2239] [Q&A] [answered]
Everything is in the title, just say yes and give a date 😂

**Comment by rrousselGit** [comment, marked as answer]
There is no plan for including one.

**Comment by julienlebren** [comment]
Could you explain why? Technically not possible, or the StateProvider is intended to disappear?

I was asking this because I have some lints `Generated providers should only depend on other generated providers` due to the fact that some of my generated providers depend on StateProviders.

**Reply by rrousselGit** [reply]
Kinda both. There's no clear syntax for stateproviders, and I feel like they are overused anyway.

Use `@riverpod class` :)

**Reply by julienlebren** [reply]
You should write a clear example of how a StateProvider can be transformed to a @riverpod class 🫣
Maybe update the _counter_ example?

**Reply by rrousselGit** [reply]
The solution is effectively "do whatever you'd do without StateProvider (so StateNotifier & co)"

**Reply by julienlebren** [reply]
Would you agree to update the most basic example (counter) with @riverpod annotation instead of `StateProvider` or do you want to keep this example independent from the @riverpod annotation?

Below the updated counter example 👇

```dart
@riverpod
class Counter extends _$Counter {
  @override
  int build() => 0;

  void increment() => state++;
}

class Home extends ConsumerWidget {
  @override
  Widget build(BuildContext context, WidgetRef ref) {
    return Scaffold(
      appBar: AppBar(title: const Text('Counter example')),
      body: Center(
        // Consumer is a widget that allows you reading providers.
        child: Consumer(
          builder: (context, ref, _) {
            final count = ref.watch(counterProvider);
            return Text('$count');
          },
        ),
      ),
      floatingActionButton: FloatingActionButton(
        // The read method is a utility to read a provider without listening to it
        onPressed: () => ref.read(counterProvider.notifier).increment(),
        child: const Icon(Icons.add),
      ),
    );
  }
}
```

**Reply by julienlebren** [reply]
I can make a PR if you agree with this kind of example.

**Reply by rrousselGit** [reply]
Sure. In general I need to update all the existing examples to the latest features/syntaxes.

**Comment by ValentinVignal** [comment]
There is a package to generated the equivalent of the `StateProvider`: 

https://pub.dev/packages/riverpod_state_provider_annotation
https://pub.dev/packages/riverpod_state_provider_generator

```dart
# main.dart
import 'package:riverpod_state_provider_annotation/riverpod_state_provider_annotation.dart';

part 'main.g.dart';
part 'main.rsp.dart';

@RiverpodStateProvider()
int myState() {
  return 0;
}

ref.watch(myStateProvider);
ref.read(myStateProvider.notifier).state++;
ref.read(myStateProvider.notifier).update((state) => state + 1);

ProviderScope(
  overrides: [
    myStateProvider.overrideWithValue(10),
  ],
)
```

It still has some limitations but for simple usages (usually the case for `StateProvider`), it does the job
